src/panels/ShareModelPanel/ShareModel.tsx

```diff
@@ -28,8 +95,26 @@ export default function ShareModel() {
     access: string;
   };
 
+  const userAlreadyHasAccess = (userName: string) => {
+    return users.some((userEntry: User) => userEntry.user === userName);
+  };
+
+  const handleRemoveUser = async (userName: string) => {
+    const userAccess: userAccess = { name: userName, access: null };
+
+    await setModelSharingPermissions(
```

Comment:
A number of these arguments never change. The only ones that do are the users access, name and, type. You could clean up calls to this by having a curried version of this function that had the previous values pre-bound.

Reply:
I think it's clearer this way... there's already enough Russian dolls in this data flow without adding another one.